fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/services/functions/UniqueValueSupplier.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.kernel.api.services.functions;
+
+import java.util.function.Supplier;
+
+/**
+ * A {@link java.util.function.Supplier} interface that guarantees the uniqueness of its provided values.
+ *
+ * @author acoburn
+ */
+public interface UniqueValueSupplier extends Supplier<String> {
```

Comment:
I put them in `o.f.k.a.services.functions` because they are different types of `Supplier` interfaces, which, itself provides a functional interface. That's the only reason.... If you'd prefer they go in `o.f.k.a.identifiers`, that's fine by me.